File: Data Structures & Algorithms/combinations/submission-1.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        res = []


        def backtrack(start, curr):
            if len(curr) == k:
                res.append(list(curr))


            for i in range(start, n+ 1):
                curr.append(i)
                backtrack(i+1, curr)
                curr.pop()
        

        backtrack(1, [])

        return res

File: Data Structures & Algorithms/daily-temperatures/submission-1.js
class Solution {
    /**
     * @param {number[]} temperatures
     * @return {number[]}
     */
    dailyTemperatures(temperatures) {
        let res = Array(temperatures.length).fill(0)
        let l = temperatures.length
        let st = []

        for (let i in temperatures){
            let curr = temperatures[i]
            while (st.length && curr > st[st.length-1][0]){
                let [_, ind] = st.pop()
                res[ind] = i - ind
            }
            st.push([curr, i])
        }

        return res
    }
}
